src/Integration/Notifications/SonarQubeNotificationService.cs

```diff
@@ -86,48 +88,53 @@ public void Stop()
 
         private async Task UpdateEvents()
         {
-            if (!sonarQubeService.IsConnected)
+            if (!sonarQubeService.IsConnected ||
+                !Model.AreNotificationsEnabled)
             {
                 return;
             }
 
-            var events = await GetNotificationEvents();
-            if (events == null)
+            try
             {
-                // Notifications are not supported on SonarQube or an exception was thrown
-                Stop();
-                return;
+                var events = await GetNotificationEvents();
+                if (events == null)
+                {
+                    // Notifications are not supported on SonarQube
+                    Stop();
+                    return;
+                }
+
+                if (!isFirstServerQuery)
```

Comment:
I don't like flags like this :) I actually don't understand what is this about :)

If the goal is to show notifications on the second call to this method, I would extract a new method - AreNotificationsEnabled() that will be called instead of UpdateEvents() in StartAsync()

Reply:
I've changed it a bit and added comments - should be more obvious now.